src/main/java/net/machpi/runelite/influxdb/InfluxDbPlugin.java

```diff
@@ -55,34 +68,62 @@ InfluxDbConfig getConfig(ConfigManager configManager) {
     @Inject
     private MeasurementCreator measurer;
 
+    @Inject
+    private ActivityState activityState;
+
     /**
      * Don't use a shared executor because we don't want to block any game threads.
      */
     private final ScheduledExecutorService executor = new ExecutorServiceExceptionLogger(Executors.newSingleThreadScheduledExecutor());
+    private final EnumMap<Skill, Integer> previousStatXp = new EnumMap<>(Skill.class);
+    private GameState prevGameState;
 
     @Subscribe
     public void onStatChanged(StatChanged statChanged) {
-        if (!config.writeXp()) {
+        int prevStatXp = previousStatXp.getOrDefault(statChanged.getSkill(), 0);
```

Comment:
This is still triggering xp updates upon login. And sending fake slayer activity.

```suggestion
        Integer prevStatXp = previousStatXp.put(statChanged.getSkill(), statChanged.getXp());
        if (prevStatXp == null || statChanged.getXp() == prevStatXp)
            return;
```